docsite/source/design-pattern.html.md

```diff
@@ -0,0 +1,36 @@
+---
+title: Design Pattern
+layout: gem-single
+name: dry-operation
+---
+
+dry-operation implements a pattern that closely resembles monadic composition, particularly the `Result` monad, and the Railway Oriented Programming pattern. Understanding these monadic concepts can provide deeper insight into how dry-operation works and why it's designed this way.
+
+### Monadic composition
+
+In functional programming, a monad is a structure that represents computations defined as sequences of steps. A key feature of monads is their ability to chain operations, with each operation depending on the result of the previous one.
+
+dry-operation emulates this monadic behavior through its `#step` method and the overall structure of operations.
+
+In monadic terms, the `#step` method in `Dry::Operation` acts similarly to the `bind` operation:
+
+1. It takes a computation that may succeed or fail (returning `Success` or `Failure`).
+1. If the computation succeeds, it extracts the value and passes it to the next step.
+1. If the computation fails, it short-circuits the entire operation, skipping subsequent steps.
+
+This behavior allows for clean composition of operations while handling potential failures at each step.
+
+By expressing this behaviour via `#step`, dry-operation lets you intermingle ordinary Ruby code in between steps as required.
```

Comment:
I added this line, since I think it's a pretty important aspect of dry-operation.

In fact, I wonder if we should make a bigger deal about this in the introduction page too? After we introduce the steps for the first time, we could have a line saying something like "steps don't need to immediately follow each other. You can add your own regular Ruby code between the steps to adjust values as required." And then you could have a little code example where something manipulates some args or something, to provide a demonstration of why this is helpful.

What do you reckon?

Reply:
I think you're totally right. Done! Let me know what you think